### PR TITLE
Get basic login functionality set up

### PR DESCRIPTION
**Pull request by himaniyadav**
Added basic login functionality and a very basic UI:
- LoginServlet.java creates and returns a link that users can use to sign in with. Uses the Users API.
- Auth.js will contain the code to handle login requests.
- Currently, there is only a sign in link. No Register/Login toggle.

Here is a screenshot of what the page currently looks like: https://screenshot.googleplex.com/gSCpPCuGEDy

### CODE REVIEW

File: capstone/src/main/java/com/google/sps/Objects/User.java [hunk 1]
```diff
@@ -4,24 +4,26 @@
...
+  private final String firstname;
   private final String email;
   private final ArrayList<Badge> badges;
-  private final ArrayList<Group> groups;
+  private final ArrayList<String> groups;
   private final ArrayList<String> interests;
 
-  public User(int userId, String name, String email, ArrayList<Badge> badges,
-      ArrayList<Group> groups, ArrayList<String> interests) {
+  public User(String userId, String firstname, String lastname, String email,
```
Comment (anikabagga) [inline capstone/src/main/java/com/google/sps/Objects/User.java]: we could add get/set methods for first name, last name, phone number, interests

Comment (g-asweet) [reply capstone/src/main/java/com/google/sps/Objects/User.java]: I would also recommend camel casing `firstName`, `lastName`

File: capstone/src/main/java/com/google/sps/Objects/User.java [hunk 2]
```diff
@@ -47,6 +71,20 @@ public String getEmail() {
...
+    this.firstName = firstName;
+    this.name = firstName + " " + getLastName();
+  }
+
+  public void setLastName (String lastName) {
+    this.lastName = lastName;
+    this.name = getFirstName() + " " + lastName;
+  }
+
+  public void setPhoneNumber(int newPhoneNumber) {
```
Comment (g-asweet) [inline capstone/src/main/java/com/google/sps/Objects/User.java]: I'm going to highly recommend against an int for phone number storage. You'll likely want a string that you format to your exact specification, as phone numbers are less dry and cut than you'd expect with area codes, international codes, etc. Additionally an int does not have enough bits to represent all phone numbers.

But it's generally agreed a string is better than a Long for this purpose
https://stackoverflow.com/questions/3483156/whats-the-right-way-to-represent-phone-numbers

**Review by g-asweet — CHANGES_REQUESTED** [review]

File: capstone/src/main/java/com/google/sps/Objects/User.java (hunk 2 above)
Comment (himaniyadav) [reply capstone/src/main/java/com/google/sps/Objects/User.java]: I see, that makes sense, thanks!

**Review by g-asweet — APPROVED** [review]

**Pull request merged**